testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergProcedureCalls.java

```diff
@@ -59,6 +60,129 @@ public void testMigrateHiveTable()
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123456'");
+
+        onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')");
+
+        Timestamp expectedTimestamp = Timestamp.valueOf("2021-01-01 10:11:12.123456");
+        assertThat(onTrino().executeQuery("SELECT x FROM " + icebergTableName))
+                .containsOnly(row(expectedTimestamp));
+        assertThat(onSpark().executeQuery("SELECT x FROM " + sparkTableName))
+                .containsOnly(row(expectedTimestamp));
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + icebergTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testMigrateTimestampNanosHiveTableWithOrc()
+    {
+        String tableName = "test_migrate_timestamp" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+        String icebergTableName = "iceberg.default." + tableName;
+        String sparkTableName = "iceberg_test.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+        onTrino().executeQuery("SET SESSION hive.timestamp_precision = 'NANOSECONDS'");
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(9)) WITH (format='ORC')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123456789'");
+
+        onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')");
+
+        assertThat(onTrino().executeQuery("SELECT x FROM " + icebergTableName))
+                .containsOnly(row(Timestamp.valueOf("2021-01-01 10:11:12.123457")));
+        assertThat(onSpark().executeQuery("SELECT x FROM " + sparkTableName))
+                .containsOnly(row(Timestamp.valueOf("2021-01-01 10:11:12.123456")));
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + icebergTableName);
+    }
+
+    @Test(groups = {ICEBERG, PROFILE_SPECIFIC_TESTS})
+    public void testMigrateTimestampMillisHiveTableWithAvro()
+    {
+        String tableName = "test_migrate_timestamp" + randomNameSuffix();
+        String hiveTableName = "hive.default." + tableName;
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+        onTrino().executeQuery("CREATE TABLE " + hiveTableName + " (x timestamp(3)) WITH (format='AVRO')");
+        onTrino().executeQuery("INSERT INTO " + hiveTableName + " VALUES timestamp '2021-01-01 10:11:12.123'");
+
+        assertThatThrownBy(() -> onTrino().executeQuery("CALL iceberg.system.migrate('default', '" + tableName + "')"))
+                .hasMessageContaining("Migrating timestamp type with AVRO format is not supported.");
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + hiveTableName);
+    }
+
+    @DataProvider(name = "timestampHiveParquet")
+    public static Object[][] parametersForTimestampMigrationOnParquet()
```

Comment:
We shouldn't use `@DataProvider`. Please take a look at https://trino.io/docs/current/develop/tests.html#avoid-data-providers-and-parametric-tests

Reply:
reimplemented